docs/src/main/asciidoc/mongodb-panache.adoc

```diff
@@ -528,7 +529,41 @@ you need to provide the value by yourself.
 ====
 
 `ObjectId` can be difficult to use if you want to expose its value in your REST service.
-So we created JSON-B and Jackson providers to serialize/deserialize them as a `String` which are are automatically registered if your project depends on one of the RESTEasy with JSON-B or RESTEasy with Jackson extensions.
+So we created JSON-B and Jackson providers to serialize/deserialize them as a `String` which are automatically registered if your project depends on one of the RESTEasy with JSON-B or RESTEasy with Jackson extensions.
+
+== Working with Kotlin Data classes
+
+Kotlin data classes are very practical to defines data carrier classes, so they are a perfect match for an entity class.
+
+But this type of class comes with some limitations: all fields needs to be initialized at construction time or be marked at nullable,
+and the generated constructor need to have as parameters all the fields of the data class.
+
+MongoDB with Panache uses the PojoCodec, this MongoDB codec mandate the presence of an empty (no parameter) constructor.
+
+So, if you want to use a data class for an entity class, you need a way to have Kotlin generates an empty constructor.
```

Comment:
```suggestion
Therefore, if you want to use a data class as an entity class, a way is needed to make Kotlin generate an empty constructor.
```

Reply:
Have you perhaps also tried the Kotlin `no-arg` compiler plugin?

Reply:
No, I can ask to test it on the original issue.